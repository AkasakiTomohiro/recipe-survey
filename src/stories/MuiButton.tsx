import React from 'react';

import { Button } from '@mui/material';

export interface MuiButtonProps {
  label: string;
  variant: 'contained' | 'outlined' | 'text';
}

export const MuiButton = ({ label, variant, ...rest }: MuiButtonProps) => 
  <Button variant={variant} {...rest}>
    {label}
  </Button>;