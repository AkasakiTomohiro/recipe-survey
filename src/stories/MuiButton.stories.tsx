import type { Meta, StoryObj } from '@storybook/react';

import { MuiButton } from './MuiButton';

const meta = {
  title: 'Example/MuiButton',
  component: MuiButton,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
  },
} satisfies Meta<typeof MuiButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    label: 'Button',
    variant: 'contained'
  },
};
