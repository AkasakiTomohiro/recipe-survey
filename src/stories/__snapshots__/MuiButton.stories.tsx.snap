// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/MuiButton Primary smoke-test 1`] = `
<button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-vbxad1-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
>
  Button
  <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
  </span>
</button>
`;
